resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/tpr/TPRCrudCallsImpl.scala

```diff
@@ -0,0 +1,320 @@
+  private val httpClient: OkHttpClient =
+    extractHttpClientFromK8sClient(k8sClient.asInstanceOf[BaseClient])
+  private val namespace: String = k8sClient.getNamespace
+  private var watchSource: BufferedSource = _
+  private lazy val buffer = new Buffer()
+
+  private implicit val formats: Formats = DefaultFormats + JobStateSerDe
+  private implicit val ec: ThreadPoolExecutor = ThreadUtils
+    .newDaemonCachedThreadPool("tpr-watcher-pool")
+
+  override def createJobObject(name: String, keyValuePairs: Map[String, Any]): Unit = {
+    val resourceObject =
+      SparkJobState(s"$TPR_API_GROUP/$TPR_API_VERSION", TPR_KIND, Metadata(name), keyValuePairs)
+    val payload = parse(write(resourceObject))
+
+    val requestBody = RequestBody
+      .create(MediaType.parse("application/json"), compact(render(payload)))
+
+    val requestSegments = Seq(
+      "apis", TPR_API_GROUP, TPR_API_VERSION, "namespaces", namespace, "sparkjobs")
+    val url = generateHttpUrl(requestSegments)
+
+    val request = completeRequest(new Request.Builder()
+      .post(requestBody)
+      .url(url))
+
+    logDebug(s"Create SparkJobResource Request: $request")
+    val response = httpClient.newCall(request).execute()
+    completeRequestWithExceptionIfNotSuccessful(
+      "post",
+      response,
+      Option(Seq(name, response.toString, compact(render(payload)))))
+
+    response.body().close()
+    logDebug(s"Successfully posted resource $name: " +
+      s"${pretty(render(parse(write(resourceObject))))}")
+  }
+
+  override def deleteJobObject(tprObjectName: String): Unit = {
+    val requestSegments = Seq(
+      "apis", TPR_API_GROUP, TPR_API_VERSION, "namespaces", namespace, "sparkjobs", tprObjectName)
+    val url = generateHttpUrl(requestSegments)
+
+    val request = completeRequest(new Request.Builder()
+      .delete()
+      .url(url))
+
+    logDebug(s"Delete Request: $request")
+    val response = httpClient.newCall(request).execute()
+    completeRequestWithExceptionIfNotSuccessful(
+      "delete",
+      response,
+      Option(Seq(tprObjectName, response.message(), request.toString)))
+
+    response.body().close()
+    logInfo(s"Successfully deleted resource $tprObjectName")
+  }
+
+  override def getJobObject(name: String): SparkJobState = {
+    val requestSegments = Seq(
+      "apis", TPR_API_GROUP, TPR_API_VERSION, "namespaces", namespace, "sparkjobs", name)
+    val url = generateHttpUrl(requestSegments)
+
+    val request = completeRequest(new Request.Builder()
+      .get()
+      .url(url))
+
+    logDebug(s"Get Request: $request")
+    val response = httpClient.newCall(request).execute()
+    completeRequestWithExceptionIfNotSuccessful(
+      "get",
+      response,
+      Option(Seq(name, response.message()))
+    )
+
+    logInfo(s"Successfully retrieved resource $name")
+    read[SparkJobState](response.body().string())
+  }
+
+  override def updateJobObject(name: String, value: String, fieldPath: String): Unit = {
```

Comment:
Based on some of the usages of this, it would be nice to have a "batch update" method that can update multiple fields at once. In doing so, we can encode updates with a POJO that's more expressive than just String fields:

`case class SparkJobPatch(field1: Option[String] = None, field2: Option[String] = None)`.

Since the fields are all optional with default values, we can be as selective as we want in what items to update. We can accomplish something similar with an `updateJobObject` method signature that takes in all of the possible updatable fields with default values, but I like encapsulating this in a single variadic argument type, so to speak.